lib/pbench/server/database/models/datasets.py

```diff
@@ -913,10 +888,9 @@ class Metadata(Database.Base):
     key = Column(String(255), unique=False, nullable=False, index=True)
     value = Column(JSON, unique=False, nullable=True)
     dataset_ref = Column(Integer, ForeignKey("datasets.id"), nullable=False)
-    user_ref = Column(Integer, ForeignKey("users.id"), nullable=True)
 
     dataset = relationship("Dataset", back_populates="metadatas", single_parent=True)
-    user = relationship("User", back_populates="dataset_metadata", single_parent=True)
+    user = Column(String(255), nullable=True)
```

Comment:
Here again, I'm concerned about "username" vs "user ID". Is a Keycloak username always guaranteed to be unique and unchangeable, or only the ID? This includes across deletions... e.g., if I create `dbutenhof` with ID 1234, delete it, can I create a new `dbutenhof` with a different ID? (Presumably, the ID would *always* be different and unique!) This would cause the stale metadata record to be "inherited" by the new user, which is bad.

Unless this is all impossible, let's stick with ID.

Reply:
> I don't think we want to store the user name in more than one place.

Exactly. That's why there's no `User` table going forward, and we're working now to reduce the join foreign key relationships to a simple string UUID column.

I think the "one place" is our backend Keycloak broker, but once we have the translation hidden behind a clean API we can finesse that.

Reply:
If the broker has the information and there's no real difference between consulting the broker and doing an SQL table lookup, then that's fine:  we can hide the translation behind an API and worry no further.

But I'm imagining a user trying to search for datasets based on the username of the owner.  I don't think we're going to be able (reliably) to convert the username into a (single) UUID, and, therefore, we won't be able to do a lookup using the UUID.  Instead, we're going want to do some operation which allows us to associate a username with each result, either by translating each dataset's owner UUID into a username (which sounds expensive) or by attaching the appropriate row from a User table to the row from the Dataset table (i.e., via a join operation), so that we can then query-filter the results.  I think that that will be much easier if we have a User table in the same database as the Dataset table, rather than having to go out to the Keycloak server for each owner (hopefully we wouldn't have to do it for each dataset!)

I suppose, rather than doing a join operation, we could do a username-to-UUID translation, and then take the _set_ of results and do an "or-search" for them.  That would be reasonably efficient, but then we could only do partial-match searches if the username-to-UUID translation supported partial-match queries.

Reply:
I don't see any need for partial username matches. We don't do it now, and it's hard to imagine that really being any more than "huh, stupid API tricks".

Matching on "groups of users" is better managed through, well, groups.

Basically SQL joins are powerful; but I doubt it gives us enormous gains for anything we're currently doing vs a separate name lookup query. (And it might even be hurting us to return a complete `User` row for every `Dataset` as we're currently doing when maybe we really only want the `username`.)

Reply:
I'm not saying that we should do it for v1.0...I just don't want us to make a choice which prevents us from doing it, unless there's a very good reason for it.